src/overview/components/Overview.jsx

```diff
@@ -1,48 +1,98 @@
+  maxWidth: 255,
+  marginRight: 'auto',
+}; 
 
 class Overview extends React.Component {
+
+  
     render() {
-        return <div>
-            <input
-                className={styles.query}
-                onChange={e=>this.props.onInputChanged(e.target.value)}
-                placeholder="Search your memory"
-                value={this.props.query}
-                ref='inputQuery'
-            >
-            </input>
-            {this.props.waitingForResults
-                ? <LoadingIndicator />
-                : <ResultList searchResult={this.props.searchResult} />
-            }
-        </div>
+     
+           const { date, onChange } = this.props;
+        return (
+      <div>
+
+               <input
+                      className={styles.query}
+                        onChange={e=>this.props.onInputChanged(e.target.value)}
+                        placeholder="Search your memory"
+                        value={this.props.query}
+                        ref='inputQuery'
+                 >
+              </input>
+                 <div>
+                  <DatePicker
+                    placeholderText="select startdate"
+                    selected={this.props.current_Start_date }
+                    minDate={moment().subtract(365,"days")}
+                    maxDate={moment()}
+                    onChange={e=>this.props.onStartDateChange(e)}
+    
+                  />
+                  <DatePicker
+                    placeholderText="select startdate"
+                    selected={this.props.current_End_date}
+                    minDate={moment().subtract(365,"days")}
+                    maxDate={moment()}
+                    onChange={e=>this.props.onEndDateChange(e)}
+                
+                  />
+                    </div>
+
+                 
+                <ResultList searchResult={this.props.searchResult} />
+              
+      </div>
+    );
+                                    
+            
     }
 
-    componentDidMount() {
+     componentDidMount() {
         if (this.props.grabFocusOnMount) {
             this.refs['inputQuery'].focus()
         }
     }
+    
 }
 
 const mapStateToProps = (state) => ({
-    query: ourState(state).query,
-    searchResult: ourState(state).searchResult,
-    waitingForResults: ourState(state).waitingForResults,
+
+      query: ourState(state).query,
+      searchResult: ourState(state).searchResult,
+      //intial setups of dates 
+      current_Start_date: ourState(state).startDate,
+      current_End_date:  ourState(state).endDate
 })
 
 const mapDispatchToProps = (dispatch) => ({
-    onInputChanged: input => {
+      onInputChanged: input => {
+       
         dispatch(actions.setQuery({query: input}))
-    },
-})
+           
+      },
+  //used for updatind the selcted date.
+      onStartDateChange: date => {
+             Myclass.Global_startDate =date;
+             dispatch(actions.handleStartChange({current_Start_date: date}))
```

Comment:
In case it was not clear to you why the names have to match, read [this introduction](https://timkendrick.co.uk/articles/2016/03/31/named-es6-arguments.html) to named arguments in ES6.